FlowCrypt/src/main/java/com/flowcrypt/email/api/email/model/ExtraActionInfo.kt

```diff
@@ -80,36 +67,33 @@ data class ExtraActionInfo constructor(
      * @param intent An incoming intent.
      */
     fun parseExtraActionInfo(context: Context, intent: Intent): ExtraActionInfo? {
-      var info: ExtraActionInfo? = null
+      var infoFromRFC6068Parser: ExtraActionInfo? = null
+      val attsList = ArrayList<AttachmentInfo>()
+      var finalBody: String? = null
+      var finalSubject: String? = null
 
       //parse mailto: URI
       if (intent.action in listOf(Intent.ACTION_VIEW, Intent.ACTION_SENDTO)) {
         if (RFC6068Parser.isMailTo(intent.data)) {
-          info = RFC6068Parser.parse(intent.data)
+          infoFromRFC6068Parser = RFC6068Parser.parse(intent.data)
         } else return null
       }
 
-      if (info == null) {
-        info = ExtraActionInfo()
-      }
-
       when (intent.action) {
         Intent.ACTION_VIEW, Intent.ACTION_SENDTO, Intent.ACTION_SEND, Intent.ACTION_SEND_MULTIPLE -> {
 
           val extraText = intent.getCharSequenceExtra(Intent.EXTRA_TEXT)
           // Only use EXTRA_TEXT if the body hasn't already been set by the mailto: URI
-          if (extraText != null && TextUtils.isEmpty(info.body)) {
-            info.body = extraText.toString()
+          if (extraText != null && infoFromRFC6068Parser?.initializationData?.body.isNullOrEmpty()) {
+            finalBody = extraText.toString()
           }
 
           val subj = intent.getStringExtra(Intent.EXTRA_SUBJECT)
           // Only use EXTRA_SUBJECT if the subject hasn't already been set by the mailto: URI
-          if (subj != null && TextUtils.isEmpty(info.subject)) {
-            info.subject = subj
+          if (subj != null && infoFromRFC6068Parser?.initializationData?.subject.isNullOrEmpty()) {
+            finalSubject = subj
```

Comment:
Instead of this  we can simply use below code and just use finalSubject in `return`

`finalSubject = infoFromRFC6068Parser?.initializationData?.subject ?: subj;`

```kt
return infoFromRFC6068Parser?.copy(
    atts = attsList,
    initializationData = infoFromRFC6068Parser.initializationData.copy(
      subject = finalSubject,
      body = finalBody,
    )
```